drivers/pinctrl/pinctrl_stm32.c

```diff
@@ -122,21 +120,37 @@ static int stm32_pins_remap(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt)
 
 	/* A valid remapping configuration is available */
 	/* Apply remapping before proceeding with pin configuration */
-	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+	if (!LL_APB2_GRP1_IsEnabledClock(LL_APB2_GRP1_PERIPH_AFIO)) {
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+	}
```

Comment:
We usually don't do that as enabling clock twice has no impact and checking systematically would has footprint impact.